### DISCUSSION TITLE
Wrong type hints on screenshots?

### DISCUSSION THREAD

**Discussion by MicaelJarniac** [fastapi/fastapi#4915] [Questions] [answered]
The code used as an example is the following:
https://fastapi.tiangolo.com/tutorial/body/#create-your-data-model
```py
from fastapi import FastAPI
from pydantic import BaseModel


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


app = FastAPI()


@app.post("/items/")
async def create_item(item: Item):
    return item
```

However, later on, on the screenshots, the following code appears:
https://fastapi.tiangolo.com/tutorial/body/#editor-support
![Screenshot](https://fastapi.tiangolo.com/img/tutorial/body/image03.png)
```py
from fastapi import FastAPI
from pydantic import BaseModel


class Item(BaseModel):
    name: str
    description: str = None
    price: float
    tax: float = None


app = FastAPI()


@app.post("/items/")
async def create_item(item: Item):
    return item
```

Notice how the type hints on `Item` are different:
```diff
 class Item(BaseModel):
     name: str
-    description: str | None = None
+    description: str = None
     price: float
-    tax: float | None = None
+    tax: float = None
```

As far as I know, the type hints on the screenshots are invalid.

**Comment by MicaelJarniac** [comment]
I just noticed that on the screenshots, there are red squiggly lines, so that makes it seem like this did give errors. Was it left this way on purpose?

**Comment by Kludex** [comment, marked as answer]
This was fixed already.
